python_modules/dagster/dagster/_seven/__init__.py

```diff
@@ -123,10 +123,76 @@ def builtin_print() -> str:
     return "builtins.print"
 
 
-def get_current_datetime_in_utc() -> Any:
+def _mockable_get_current_datetime_in_utc():
+    # Can be mocked in tests by freeze_time()
     return datetime.now(tz=timezone.utc)
 
 
+def get_current_datetime_in_utc() -> Any:
+    return _mockable_get_current_datetime_in_utc()
+
+
+def _mockable_get_current_timestamp():
+    return time.time()
+
+
+def get_current_timestamp():
+    # Like time.time() but can be mocked in tests by freeze_time()
+    return _mockable_get_current_timestamp()
+
+
+def create_utc_datetime(year, month, day, *args, **kwargs):
+    return datetime(year, month, day, *args, **kwargs, tzinfo=timezone.utc)
+
+
+def add_fixed_time(
+    dt: datetime,
+    *,
+    hours=0,
+    minutes=0,
+    seconds=0,
+    milliseconds=0,
+    microseconds=0,
+):
+    """Behaves like adding a time using a timedelta, but handles fall DST transitions correctly
+    without skipping an hour ahead.
+    """
+    return (
+        dt.astimezone(timezone.utc)
+        + timedelta(
+            seconds=seconds,
+            microseconds=microseconds,
+            milliseconds=milliseconds,
+            minutes=minutes,
+            hours=hours,
+        )
+    ).astimezone(dt.tzinfo)
+
+
+def subtract_fixed_time(
+    dt: datetime,
+    *,
+    hours=0,
+    minutes=0,
+    seconds=0,
+    milliseconds=0,
+    microseconds=0,
+):
+    """Behaves like adding a time using a timedelta, but handles fall DST transitions correctly
+    without skipping an hour behind.
+    """
+    return (
+        dt.astimezone(timezone.utc)
+        - timedelta(
+            seconds=seconds,
+            microseconds=microseconds,
+            milliseconds=milliseconds,
+            minutes=minutes,
+            hours=hours,
+        )
+    ).astimezone(dt.tzinfo)
+
+
```

Comment:
worth gathering all this stuff in `dagster._time` or equivalent ? `seven` as a play on `six` the python 2 / 3 compat layer is increasingly out of touch the further we get from the death of 2.7

Reply:
I like dagster._time, once this stack settles I will move things out of _seven and into _time